eng/code-quality-reports/src/main/java/com/azure/customcheckstyles/ServiceClientSubclassCheck.java

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.customcheckstyles;
+
+import com.azure.common.ServiceClient;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import java.util.Optional;
+
+/**
+ * Used to check that descendants of ServiceClient follow a few rules
+ *
+ * 1) They cannot have public or protected constructors
+ * 2) They must implement a public static builder method
+ */
+public class ServiceClientSubclassCheck extends AbstractCheck {
```

Comment:
I would suggest a better name might be 'ServiceClientChecks' - and then bake in all service client checks into this one class (that would be 4 and 5 from the spec doc)